### PR TITLE
Fixed Point Multiplication and Division

### PR DESCRIPTION
**Pull request by equationcrunchor**
Also includes some changes to how the simulator handles -0 and the halting, to better match the behavior of the original machine.

Supersedes pull request #250 .

### CODE REVIEW

File: static/js/simulator.js [hunk 1]
```diff
@@ -596,19 +600,19 @@ class General_Word extends Word {
     }
 
     /**
-     * Stores a fixed point number into the word.
+     * Stores a fixed point number into the word.  Be sure to distinguish positive and negative zero.
      *
      * @param {number} number       Number to be stored.
      */
     set fixed_point(number) {
         var sign_bit = "";
```
Comment (ryaanahmed) [inline static/js/simulator.js]: prefer:
`let sign_bit;` - generally better practice to declare the variable without initializing than init to empty string, and let's avoid `var` for its odd scoping behaviors — if you could go through and generally clean up use of var and empty string initialization throughout, that would be great!

File: static/js/simulator.js [hunk 2]
```diff
@@ -1361,6 +1364,104 @@ function LDQ(computer) {
...
+ * Emulates the IBM 704 Multiply (MPY) operation.
+ *
+ * Multiplies the fixed-point value of the storage register by the fixed-point value of the Multiplier-
+ * Quotient Register.  The 35 most significant bits are stored in the Accumulator, and the 35 least
+ * significant bits are placed in the Multiplier-Quotient Register.
+ *
+ * @param {IBM_704} computer    Machine to execute instruction on.
+ */
+function MPY(computer) {
+    let result = computer.storage_register.fixed_point*computer.mq_register.fixed_point;
```
Comment (ryaanahmed) [inline static/js/simulator.js]: `const` and please always space infix operators (around `*` here) for readability

**Comment by ryaanahmed** [comment]
By the way—it's looking to me like the various computing routines should probably be methods of the `IBM_704` class, since every single one of them takes `computer` as its first parameter.

File: static/js/simulator.js (hunk 1 above)
Comment (equationcrunchor) [reply static/js/simulator.js]: Question: Can I use empty strings in a case like in the function `binary_add` (line 161), where I concatenate to the string in a loop?

**Comment by equationcrunchor** [comment]
> By the way—it's looking to me like the various computing routines should probably be methods of the `IBM_704` class, since every single one of them takes `computer` as its first parameter.

So right now the `step` method works by getting the global functions from the dictionary.  What's the best way to integrate the functions into the IBM 704 class?

File: static/js/simulator.js (hunk 1 above)
Comment (ryaanahmed) [reply static/js/simulator.js]: Yes—I don't want to be draconian about this principle. In the case of `binary_add`, declaring but not initializing the variable would require you to add some logic in the for loop (checking if it's `undefined` to catch the case of the first run), which would make the loop harder to read and maintain. So, in this case, I think initializing to the empty string is totally fine and, in fact, better.

The more general case is that defining a variable without initializing it lets us make a general assumption about our codebase that we use `=== undefined` to see if we have a value, whereas getting in the habit of `let foo = ""` means we end up with other falsey states to deal with. While it doesn't actually cause problems in any of the instances that you wrote it in, I'd like to develop the habit in this codebase of this practice, which will simplify things in the long term.

Javascript is so permissive that we have to enforce disciplines internally about this kind of stuff to keep from going mad.

**Comment by ryaanahmed** [comment]
> So right now the step method works by getting the global functions from the dictionary. What's the best way to integrate the functions into the IBM 704 class?

Ah, I see—a little tricky. Let's leave this structure in place now and take a look at it together when I'm back in the lab next week. The current structure is certainly not blocking further work, so I'm fine with leaving it be until a later stage.

**Pull request merged**